docs: add NVIDIA GPU Operator guide (#10018)
spectrocloud/librarium: docs/docs-content/ai-workloads/nvidia-gpu-operator.md

---
sidebar_label: "NVIDIA GPU Operator Pack"
title: "Enable AI Workloads with the NVIDIA GPU Operator Pack"
description:
  "Add the nvidia-gpu-operator-ai pack to a Palette cluster profile to simplify GPU infrastructure management and enable
  AI workloads on your cluster nodes."
tags: ["gpu", "nvidia", "ai workloads", "cuda", "packs", "gpu operator"]
sidebar_position: 10
---

> You can use the [NVIDIA GPU Operator](https://docs.spectrocloud.com/integrations/packs/?pack=nvidia-gpu-operator-ai)

Reviewer: ```suggestion
You can use the <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" />
```

pack to simplify the provisioning and management of GPU infrastructure for AI workloads on your Palette-managed

> clusters. This pack bundles the GPU Operator alongside supporting components for driver management, node labeling, and

Reviewer: ```suggestion
clusters. This pack bundles the [GPU Operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html) alongside supporting components for driver management, node labeling, and
```

monitoring. Follow this guide to add the pack to a cluster profile and verify that GPU workloads can run on your nodes.

## Prerequisites

> - A Palette account with permissions to create and manage cluster profiles and clusters.

Reviewer: I would link to https://docs.spectrocloud.com/user-management/palette-rbac/resource-scope-roles-permissions for additional info.

> - The `nvidia-gpu-operator-ai` pack (version 25.10.x) available in your Palette registry.

Reviewer: ```suggestion
- <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack version 25.10.0 or later available in a Palette-registered registry. Refer to [link] for information on adding registries to Palette. 
```

And then maybe link to https://docs.spectrocloud.com/registries-and-packs/registries/

> - A cluster deployed on VMs with GPU passthrough, VMs with NVIDIA vGPUs, or bare metal.

Reviewer: ```suggestion
- A cluster deployed through Palette that contains VMs with GPU passthrough enabled, VMs with NVIDIA vGPUs, or bare metal devices with GPUs.
```
Maybe? I'm assuming.

Reviewer: The guide could be a bit more clear for how we are guiding the user experience. For example, it looks like we have a prerequisite that says they need an existing cluster. (While not necessarily true, it does make the guide smaller and more straightforward.) Then later, we have a deploy cluster step, which seemingly contradicts this prerequisite.

Author: ```suggestion
- A Palette-managed cluster with nodes that are VMs with GPU passthrough enabled, VMs with NVIDIA vGPUs, or bare metal servers with GPUs
```

> - At least one node equipped with a discrete NVIDIA GPU. Embedded products such as NVIDIA Jetson are not supported.

Reviewer: ```suggestion
- At least one node equipped with a dedicated NVIDIA graphics card (discrete GPU). Embedded products such as NVIDIA Jetson are not supported.
```

> - A supported operating system and Kubernetes version. Refer to the

Reviewer: ```suggestion
- A supported OS and Kubernetes version. Refer to the
```

  [NVIDIA GPU Operator Platform Support](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/platform-support.html)
  page for details.

> - [containerd](https://containerd.io/) or [CRI-O](https://cri-o.io/) container runtime.

Reviewer: Installed on the cluster? Does it have to be containerd or CRI-O? Can it be any CRI-compatible container runtime? If the later, doesn't that already have to exist in order for the cluster to even function/run containers?

Author: On second thought, [any Kubernetes version supported by the GPU Operator](https://github.com/spectrocloud/librarium/pull/10018/changes#diff-6b2bb541af872fe71c221ccd687d99cda14d98b8922031ddc2572f998abe182bR29-R31) is guaranteed to have containerd or CRI-O. Removing this bullet as redundant.

![NVIDIA GPU Operator architecture](/ai-workloads/nvidia-gpu-operator_architecture.webp)

> ## Add the Pack to a Cluster Profile

Reviewer: ```suggestion
## Create Add-on Cluster Profile
```

This more accurately describes the process. (They have the option to add it to a full cluster profile, but the process you initially described was about creating a separate add-on profile.)

> 1. Log in to [Palette](https://console.spectrocloud.com) and select **Profiles** from the left main menu.
> 
> 2. Create a new cluster profile or open an existing one.
> 
> 3. Select **Add New Pack**.
> 
> 4. Search for `nvidia gpu operator` and select the **Nvidia GPU Operator** pack.
> 
> 5. Select pack version **25.10.x**.
> 
> 6. Review the pack values. The defaults work for most deployments. For customization options and additional bundled
>    capabilities, refer to the
>    [pack page](https://docs.spectrocloud.com/integrations/packs/?pack=nvidia-gpu-operator-ai).
> 
> 7. Select **Confirm & Create** to return to the cluster profile overview page, and then select **Save Changes**.

Reviewer: ```suggestion
1. Log in to [Palette](https://console.spectrocloud.com).

2. From the left main menu, select **Profiles**.

3. From the **Cluster Profiles** table, select **Add Cluster Profile**. 

     :::tip
     
     Alternatively, if your GPU-enabled cluster was deployed using a [full cluster profile], you can select it and add the NVIDIA GPU Operator as an additional pack to your cluster profile stack. We recommend [creating a new cluster profile version] before making any changes. 
     
     :::

4. Enter a unique **Name** for the cluster profile and choose a cluster profile **Type** of *Add-on**. Select **Next**.

5. On the **Profile Layers** step, select **Add New Pack**.  

4. Search for `nvidia gpu operator` and select the **Nvidia GPU Operator** pack.

5. Select a **Pack Version** of **25.10.0** or later.

6. Review the pack **Values**. The default values and parameters are sufficient for most deployments. For  additional customization options and bundled
   capabilities, refer to the <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack page.

7. Select **Confirm & Create** to return to the **Profile Layers** page. Select **Next**.

8. Review your cluster profile. If no changes are needed, select **Finish Configuration**.
```

Reviewer: Potential links to include:
- https://docs.spectrocloud.com/profiles/cluster-profiles/modify-cluster-profiles/version-cluster-profile/
- https://docs.spectrocloud.com/profiles/cluster-profiles/create-cluster-profiles/create-full-profile/

> ## Deploy the Cluster
> 
> Attach the profile to a new or existing cluster, then deploy or update the cluster. Initial driver compilation on GPU
> nodes can take several minutes.

Reviewer: ```suggestion
## Add Profile to Cluster

Once you have created an add-on profile, you can attach it to your GPU-enabled cluster. If you added the NVIDIA GPU Operator pack to an existing full profile, you can update the cluster profile version to the version that includes your changes. The initial driver compilation on GPU nodes can take several minutes. Refer to [Update a Cluster] for more information.

```

Suggested link: https://docs.spectrocloud.com/clusters/cluster-management/cluster-updates/

> ## Validate

Reviewer: Little introduction here to the two ways we will be verifying? :)

> 
> ### Verify GPU Node Labels
> 
> Filter for nodes labeled with `pci-10de.present=true`, which Node Feature Discovery (NFD) applies to any node where it
> detects an NVIDIA Peripheral Component Interconnect (PCI) device.

Reviewer: ```suggestion

### Verify GPU Node Labels

To verify your cluster is ready to run GPU workloads, verify the cluster nodes have the label `pci-10de.present=true`. This filter is automatically applied using Node Feature Discovery (NFD) when a NVIDIA Peripheral Component Interconnect (PCI) device is detected. 
```

Reviewer: Why do we use the label `feature.node.kubernetes.io/pci-10de.present=true` in the validate steps but shorten it to `pci-10de.present=true` here?

Author: That wasn't intentional, good catch

> 

Reviewer: ```suggestion
1. Log in to [Palette](https://console.spectrocloud.com/).

2. From the left main menu, select **Clusters**.

3. Choose your GPU-enabled cluster from the **Clusters** table.

4. From the cluster **Overview** tab, download the **Admin Kubeconfig File**. Refer to our [Kubeconfig] and [Kubectl] guides for additional information.

5. Open a terminal session in an environment that has network access to the cluster. Set the `KUBECONFIG` environment variable to the file path of the downloaded kubeconfig file.

export KUBECONFIG=<path-to-kubeconfig>

6. Search for nodes with the label `feature.node.kubernetes.io/pci-10de.present=true`. 
```

Reviewer: We need to give them guidance on how to connect to the clister.

Suggested links for suggestion ^ 
- https://docs.spectrocloud.com/clusters/cluster-management/palette-webctl/
- https://docs.spectrocloud.com/clusters/cluster-management/kubeconfig/

(ensure the command is placed in an indented codeblock as well -- git is weird with nested codeblocks, so I didn't even attempt)

```bash
kubectl get nodes --selector feature.node.kubernetes.io/pci-10de.present=true
```

> GPU nodes should appear in the output.

Reviewer: I think some example output here would be great 🙏

Confirm that the node advertises allocatable GPUs.

```bash
kubectl describe node <gpu-node-name> | grep nvidia.com/gpu
```

The value of `nvidia.com/gpu` indicates the number of GPUs available for scheduling on that node.

> ```text hideClipboard

Reviewer: ```suggestion
```text hideClipboard title="Example output"
```

nvidia.com/gpu:  1
```

### Run a GPU Workload Test

> 

Reviewer: Brief description of the test?

> Save the following pod manifest to a file named `gpu-test.yaml`.

Reviewer: ```suggestion
1. Save the following pod manifest to a file named `gpu-test.yaml`.
```

```yaml title="gpu-test.yaml"

> apiVersion: v1
> kind: Pod
> metadata:
>   name: gpu-test
> spec:
>   restartPolicy: OnFailure
>   containers:
>     - name: cuda-vector-add
>       image: "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda11.2.1"
>       resources:
>         limits:
>           nvidia.com/gpu: 1

Reviewer: ```suggestion
cat << EOF > gpu-test.yaml
apiVersion: v1
kind: Pod
metadata:
  name: gpu-test
spec:
  restartPolicy: OnFailure
  containers:
    - name: cuda-vector-add
      image: "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda11.2.1"
      resources:
        limits:
          nvidia.com/gpu: 1
EOF
```

Heredoc format to reduce the number of steps.

```

> 

Reviewer: ```suggestion
2. Apply the `gpu-test.yaml` manifest to the cluster. 
```

```bash
kubectl apply --filename gpu-test.yaml
```

> Wait for the pod to complete.

Reviewer: ```suggestion
3. Wait for the `gpu-test` pod to be deployed.
```

```bash
kubectl wait --for=jsonpath='{.status.phase}'=Succeeded pod/gpu-test --timeout=120s
```

> Then retrieve the logs.

Reviewer: ```suggestion
4. Retrieve the logs from the `gpu-test` pod to ensure the test was successful.
```

```bash
kubectl logs gpu-test
```

> ```text hideClipboard title="Expected output"

Reviewer: ```suggestion
```text hideClipboard title="Example output"
```

[Vector addition of 50000 elements]
Copy input data from the host memory to the CUDA device
CUDA kernel launch with 196 blocks of 256 threads
Copy output data from the CUDA device to the host memory
Test PASSED
Done
```

> Clean up the test pod.

Reviewer: ```suggestion
5. Delete the `gpu-test` pod.
```

```bash
kubectl delete pod gpu-test
```

## Next Steps

Now that GPU workloads can run on your cluster, you can review the
[pack page](https://docs.spectrocloud.com/integrations/packs/?pack=nvidia-gpu-operator-ai) for the full list of
configurable values.

For a deeper reference on the operator itself, including advanced configuration and known limitations, refer to the
[NVIDIA GPU Operator documentation](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html).

If you are building AI platforms on top of GPU-enabled Palette infrastructure, refer to

> [PaletteAI](https://docs.palette-ai.com/) for deploying and managing AI workloads on GPU-enabled clusters.

Reviewer: ```suggestion
Once you have verified GPU workloads can run on your cluster, you can review the
<VersionedLink text="NVIDIA GPU Operator" url="integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack page for a full list of configurable values and make the necessary modifications to fit your needs.

For additional information on the GPU Operator itself, including advanced configurations and known limitations, refer to the
[NVIDIA GPU Operator documentation](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html).

If you plan on regularly running GPU-intensive applications on clusters, such as AI/ML applications and models, consider using [PaletteAI](https://docs.palette-ai.com/) to streamline the deployment and management of GPU-enabled clusters and create predefined, reusable application bundles that application engineers use to deploy workloads as needed. 
```

---
sidebar_label: "NVIDIA GPU Operator Pack"
title: "Enable AI Workloads with the NVIDIA GPU Operator Pack"
description:
  "Add the nvidia-gpu-operator-ai pack to a Palette cluster profile to simplify GPU infrastructure management and enable
  AI workloads on your cluster nodes."
tags: ["gpu", "nvidia", "ai workloads", "cuda", "packs", "gpu operator"]
sidebar_position: 10
---

You can use the

> [`nvidia-gpu-operator-ai`](https://docs.spectrocloud.com/integrations/packs/?pack=nvidia-gpu-operator-ai) pack to

Reviewer: ```suggestion
[NVIDIA GPU Operator](https://docs.spectrocloud.com/integrations/packs/?pack=nvidia-gpu-operator-ai) pack to
```
Packs are weird: if we reference the pack name from the cluster profile perspective (telling them what to add from the UI, for example), we use the full pack name in monospace. If we are referencing the pack in general, we just reference it by the pack title.

Author: Thank you for clarifying, makes sense

This pack bundles the [GPU Operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html)
alongside supporting components for driver management, node labeling, and monitoring. Follow this guide to add the pack
to a cluster profile and verify that GPU workloads can run on your nodes.

## Prerequisites

- A Palette account with permissions to create and manage cluster profiles and clusters.

> - <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack version 25.10.0 or later available in a Palette-registered registry. Refer to [link] for information on adding registries to Palette. 

Author: [link] to https://docs.spectrocloud.com/registries-and-packs/registries/

---
sidebar_label: "NVIDIA GPU Operator Pack"
title: "Enable AI Workloads with the NVIDIA GPU Operator Pack"
description:
  "Add the nvidia-gpu-operator-ai pack to a Palette cluster profile to simplify GPU infrastructure management and enable
  AI workloads on your cluster nodes."
tags: ["gpu", "nvidia", "ai workloads", "cuda", "packs", "gpu operator"]
sidebar_position: 10
---

> You can use the <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack
> to simplify the provisioning and management of GPU infrastructure for AI workloads on your Palette-managed clusters.
> This pack bundles the [GPU Operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html)
> alongside supporting components for driver management, node labeling, and monitoring. Follow this guide to add the pack
> to a cluster profile and verify that GPU workloads can run on your nodes.

Reviewer: ```suggestion

<!-- prettier-ignore-start -->

You can use the <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack
to simplify the provisioning and management of GPU infrastructure for AI workloads on your Palette-managed clusters.
This pack bundles the [GPU Operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html)
alongside supporting components for driver management, node labeling, and monitoring. Follow this guide to add the pack
to a cluster profile and verify that GPU workloads can run on your nodes.

<!-- prettier-ignore-end -->

```

## Prerequisites

- A Palette account with permissions to create and manage cluster profiles and clusters. Refer to

>   <VersionedLink text="Roles and Permissions" url="/user-management/palette-rbac/resource-scope-roles-permissions" />

Reviewer: No need to use a versionedLink here. We only use those for packs. You can change this to a relative path or absolute path. (We use relative ones for normal md files and absolute for partials/mdx)

  for details.
- <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack version

>   25.10.0 or later available in a Palette-registered registry. Refer to [link] for information on adding registries to

Reviewer: Missed the `[link]` here. (Sorry, I put placeholders because I did not have the relative links.) Will need to a relative link that resolves to https://docs.spectrocloud.com/registries-and-packs/registries/.

Reviewer: Also need `<!-- prettier-ignore-start -->` and `<!-- prettier-ignore-end -->` comments around this bullet.

  Palette.
- A Palette-managed cluster with nodes that are VMs with GPU passthrough enabled, VMs with NVIDIA vGPUs, or bare metal

>   servers with GPUs. If you don't have a cluster yet, refer to

Reviewer: ```suggestion
  servers with GPUs. If you do not have a cluster yet, refer to
```

>   <VersionedLink text="Clusters" url="/clusters/" /> for deployment guides.

Reviewer: Convert to relative link.

Reviewer: This is a partial that maybe be useful :) Here is an example for how it is used: https://docs.spectrocloud.com/cluster-templates/create-cluster-templates/#next-steps

`_partials/clusters/_cluster-deployment-guides.mdx`

Author: Sticking with an inline link to /clusters/ to keep the prerequisites readable. The partial renders a full table that doesn't embed well in a bullet list.

- At least one node equipped with a dedicated NVIDIA GPU. Embedded products such as NVIDIA Jetson are not supported.
- A supported OS and Kubernetes version. Refer to the
  [NVIDIA GPU Operator Platform Support](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/platform-support.html)
  page for details.

> ![NVIDIA GPU Operator architecture](/ai-workloads/nvidia-gpu-operator_architecture.webp)

Reviewer: I missed this the first time, but a more appropriate placement would be after the introduction rather than in the prerequisites.

## Create Add-on Cluster Profile

1. Log in to [Palette](https://console.spectrocloud.com).

2. From the left main menu, select **Profiles**.

3. From the **Cluster Profiles** table, select **Add Cluster Profile**.

   :::tip

   Alternatively, if your GPU-enabled cluster was deployed using a

   <VersionedLink
     text="full cluster profile"
     url="/profiles/cluster-profiles/create-cluster-profiles/create-full-profile/"
   />
   , you can select it and add the NVIDIA GPU Operator as an additional pack to your cluster profile stack. We recommend
   <VersionedLink
     text="creating a new cluster profile version"
     url="/profiles/cluster-profiles/modify-cluster-profiles/version-cluster-profile/"
   />
   before making any changes.

>    :::

Reviewer: VersionedLinks. Also, take a look at the preview. 

<img width="960" height="335" alt="Image" src="https://github.com/user-attachments/assets/9f1b0dab-02e7-4fdc-ad83-eead71d4fafe" />

4. Enter a unique **Name** for the cluster profile and choose a cluster profile **Type** of **Add-on**. Select **Next**.

5. On the **Profile Layers** step, select **Add New Pack**.

6. Search for `nvidia gpu operator` and select the **Nvidia GPU Operator** pack.

7. Select a **Pack Version** of **25.10.0** or later.

8. Review the pack **Values**. The default values and parameters are sufficient for most deployments. For additional
   customization options and bundled capabilities, refer to the

>    <VersionedLink text="NVIDIA GPU Operator" url="/integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack page.

Reviewer: VersionedLink > Relative link

Author: Pack link, therefore `<VersionedLink>` required? I did add `prettier-ignore` comments around it though.

9. Select **Confirm & Create** to return to the **Profile Layers** page. Select **Next**.

10. Review your cluster profile. If no changes are needed, select **Finish Configuration**.

## Add Profile to Cluster

Once you have created an add-on profile, attach it to your GPU-enabled cluster. If you added the NVIDIA GPU Operator
pack to an existing full profile, update the cluster profile version to the version that includes your changes. The
initial driver compilation on GPU nodes can take several minutes. Refer to

> <VersionedLink text="Update a Cluster" url="/clusters/cluster-management/cluster-updates/" /> for more information.

Reviewer: VersionedLink > Relative link

## Validate

Confirm the GPU Operator is running correctly by verifying that GPU nodes are labeled and that a GPU workload can run
successfully on the cluster.

### Verify GPU Node Labels

To verify your cluster is ready to run GPU workloads, verify the cluster nodes have the label
`feature.node.kubernetes.io/pci-10de.present=true`. This filter is automatically applied using Node Feature Discovery
(NFD) when a NVIDIA Peripheral Component Interconnect (PCI) device is detected.

1. Log in to [Palette](https://console.spectrocloud.com/).

2. From the left main menu, select **Clusters**.

3. Choose your GPU-enabled cluster from the **Clusters** table.

4. Download the kubeconfig file for your cluster and configure kubectl to use it. Refer to

>    <VersionedLink text="Access Cluster with CLI" url="/clusters/cluster-management/kubeconfig/" /> for guidance.

Reviewer: Versioned > relative link

>    ```bash

Reviewer: Indentation is off throughout this area. Codeblocks need to be nested under steps. Run `make start` locally while you are indenting the code blocks to make sure everything is correct 💪 If you haven't already, consider installing prettier locally to help: https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/3142057986/Prettier

<img width="1066" height="775" alt="Image" src="https://github.com/user-attachments/assets/8b85e76e-fa07-4026-b5cb-d08ef201288d" />

   export KUBECONFIG=<path-to-kubeconfig>
   ```

5. Search for nodes with the label `feature.node.kubernetes.io/pci-10de.present=true`.

   ```bash
   kubectl get nodes --selector feature.node.kubernetes.io/pci-10de.present=true
   ```

   GPU nodes should appear in the output.

   ```text hideClipboard title="Example output"
   NAME                    STATUS   ROLES    AGE   VERSION
   gpu-worker-node-name    Ready    <none>   11m   v1.33.6
   ```

> 6. Confirm that the node advertises allocatable GPUs.

Reviewer: ```suggestion
6. Confirm that the node advertises allocatable GPUs. Replace `<gpu-node-name>` with the GPU node. 
```

   ```bash
   kubectl describe node <gpu-node-name> | grep nvidia.com/gpu
   ```

   The value of `nvidia.com/gpu` indicates the number of GPUs available for scheduling on that node.

   ```text hideClipboard title="Example output"
   nvidia.com/gpu:  1
   ```

> ### Run a GPU Workload Test

Reviewer: ```suggestion
### Run GPU Workload Test
```

> 1. Deploy a test pod using the standard NVIDIA CUDA vector addition sample to confirm the node can run GPU workloads.

Reviewer: Indentation is off throughout this section as well 

<img width="983" height="1077" alt="Image" src="https://github.com/user-attachments/assets/4f6cfd1c-43da-4689-9694-9ed4532915d6" />

> 1. Deploy a test pod using the standard NVIDIA CUDA vector addition sample to confirm the node can run GPU workloads.

Reviewer: ```suggestion
1. Deploy a test pod using the standard [NVIDIA CUDA vector addition sample](https://github.com/olcf-tutorials/vector_addition_cuda) to confirm the node can run GPU workloads.
```

   ```bash
   cat << EOF | kubectl apply --filename -
   apiVersion: v1
   kind: Pod
   metadata:
     name: gpu-test
   spec:
     restartPolicy: OnFailure
     containers:
       - name: cuda-vector-add
         image: "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda11.2.1"
         resources:
           limits:
             nvidia.com/gpu: 1
   EOF
   ```

> 2. Wait for the pod to complete.

Reviewer: ```suggestion
2. Wait for the pod to be provisioned.
```

Reviewer: "Complete" is a transitive verb that usually needs an object. "Complete what?" Passive voice is OK here because we are waiting for a result.

   ```bash
   kubectl logs gpu-test
   ```

   ```text hideClipboard title="Example output"
   [Vector addition of 50000 elements]
   Copy input data from the host memory to the CUDA device
   CUDA kernel launch with 196 blocks of 256 threads
   Copy output data from the CUDA device to the host memory
   Test PASSED
   Done
   ```

4. Clean up the test pod.

   ```bash
   kubectl delete pod gpu-test
   ```

## Next Steps

Once you have verified GPU workloads can run on your cluster, you can review the

<VersionedLink text="NVIDIA GPU Operator" url="integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack page for a full
list of configurable values and make the necessary modifications to fit your needs.

For additional information on the GPU Operator itself, including advanced configurations and known limitations, refer to
the [NVIDIA GPU Operator documentation](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html).

If you plan on regularly running GPU-intensive applications on clusters, such as AI/ML applications and models, consider
using [PaletteAI](https://docs.palette-ai.com/) to streamline the deployment and management of GPU-enabled clusters and

> create predefined, reusable application bundles that application engineers use to deploy workloads as needed.

Reviewer: ```suggestion
<!-- prettier-ignore-start -->

Once you have verified GPU workloads can run on your cluster, you can review the <VersionedLink text="NVIDIA GPU Operator" url="integrations/packs/?pack=nvidia-gpu-operator-ai" /> pack page for a full list of configurable values and make the necessary modifications to fit your needs. For additional information on the GPU Operator itself, including advanced configurations and known limitations, refer to the [NVIDIA GPU Operator documentation](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/index.html).

<!-- prettier-ignore-end -->

If you plan on regularly running GPU-intensive applications on clusters, such as AI/ML applications and models, consider
using [PaletteAI](https://docs.palette-ai.com/) to streamline the deployment and management of GPU-enabled clusters and
create predefined, reusable application bundles that application engineers use to deploy workloads as needed.

```
